src/qt3utils/datagenerators/spectrometers/princeston.py

```diff
@@ -228,100 +213,111 @@ def experiment_name(self, a_name: str) -> None:
+    def sensor_temperature_set_point(self, deg_celsius: float) -> None:
         """
-        Sets the number of frames to be taken during acquisition to number.
+        This function retrieves image data, with particular attention to the camera's configuration determined by the `sensor_temperature_set_point`.
+        The `sensor_temperature_set_point` defines a target or reference value for the camera's sensor, ensuring optimal or specific operation conditions for image acquisition.
+        Depending on the setpoint, the behavior or response of the camera sensor might vary.
         """
-        self.light.set(lf.AddIns.ExperimentSettings.AcquisitionFramesToStore, Int64(num_frames))
+        self.light.set(lf.AddIns.CameraSettings.SensorTemperatureSetPoint, Double(deg_celsius))
 
     @property
     def exposure_time(self) -> float:
-        """
-        Returns the single frame exposure time (in ms).
-        """
         return self.light.get(lf.AddIns.CameraSettings.ShutterTimingExposureTime)
 
     @exposure_time.setter
     def exposure_time(self, ms: float) -> None:
-        """
-        Sets the single frame exposure time to be ms (in ms).
-        """
         self.light.set(lf.AddIns.CameraSettings.ShutterTimingExposureTime, Double(ms))
 
     @property
-    def sensor_temperature_set_point(self) -> float:
+    def num_frames(self) -> int:
         """
-        Returns the sensor setpoint temperature (Celsius).
+        Returns the number of frames taken during the acquisition.
         """
-        return self.light.get(lf.AddIns.CameraSettings.SensorTemperatureSetPoint)
+        return self.light.get(lf.AddIns.ExperimentSettings.AcquisitionFramesToStore)
 
-    @sensor_temperature_set_point.setter
-    def sensor_temperature_set_point(self, deg_C: float) -> None:
+    @num_frames.setter
+    def num_frames(self, num_frames: int) -> None:
         """
-        Sets the sensor target temperature (in degrees Celsius) to deg_C.
-        This function retrieves image data, with particular attention to the camera's configuration determined by the `sensor_temperature_set_point`.
-        The `sensor_temperature_set_point` defines a target or reference value for the camera's sensor, ensuring optimal or specific operation conditions for image acquisition.
-        Depending on the setpoint, the behavior or response of the camera sensor might vary.
+        Sets the number of frames to be taken during acquisition to number.
         """
-        self.light.set(lf.AddIns.CameraSettings.SensorTemperatureSetPoint, Double(deg_C))
+        self.light.set(lf.AddIns.ExperimentSettings.AcquisitionFramesToStore, Int64(num_frames))
 
-class SpectrometerDataAcquisition(SpectrometerConfig):
-    def acquire_frame(self) -> Tuple[np.ndarray, np.ndarray]:
-        """
-        Acquires a frame (or series of frames) from the spectrometer, and the
-        corresponding wavelength data.
-        """
+    def acquire(
```

Comment:
Could you please explain what is going on with this to me? I don't mind if it is lengthy, just need to fully understand the thought process

Reply:
I wrote a bit more about it on the class docstring.